front/tests/pages/op-route-page-model.ts

```diff
@@ -0,0 +1,327 @@
+  async clickOnViaAddButtons(...viaNames: string[]) {
+    for (const viaName of viaNames) {
+      await clickWithDelay(this.getAddButtonLocatorByViaName(viaName));
+      await expect(this.getDeleteButtonLocatorByViaName(viaName)).toBeVisible();
+    }
+  }
+
+  // Verifies that the specified markers are visible on the map.
+  async verifyMapMarkers(...markerNames: string[]) {
+    for (const markerName of markerNames) {
+      await expect(this.getMapPathfindingMarker(markerName)).toBeVisible();
+    }
+  }
+
+  // Validates the waypoint suggestions by checking the name, CH, UIC, and distance.
+  static async validateWaypointSuggestions(
+    waypointSuggestion: Locator,
+    expectedName: string,
+    expectedCh: string,
+    expectedUic: string,
+    expectedKm: string
+  ) {
+    await expect(RoutePage.getWaypointSuggestionNameLocator(waypointSuggestion)).toHaveText(
+      expectedName
+    );
+    await expect(RoutePage.getWaypointSuggestionChLocator(waypointSuggestion)).toHaveText(
+      expectedCh
+    );
+    await expect(RoutePage.getWaypointSuggestionUicLocator(waypointSuggestion)).toHaveText(
+      expectedUic
+    );
+    await expect(RoutePage.getWaypointSuggestionDistanceLocator(waypointSuggestion)).toHaveText(
+      expectedKm
+    );
+  }
+
+  // Validates the added waypoints by checking the name, CH, and UIC.
+  static async validateAddedWaypoint(
+    droppedWaypoint: Locator,
+    expectedName: string,
+    expectedCh: string,
+    expectedUic: string
+  ) {
+    await expect(RoutePage.getWaypointDroppedNameLocator(droppedWaypoint)).toHaveText(expectedName);
+    await expect(RoutePage.getWaypointDroppedChLocator(droppedWaypoint)).toHaveText(expectedCh);
+    await expect(RoutePage.getWaypointDroppedUicLocator(droppedWaypoint)).toHaveText(expectedUic);
+  }
+
+  // Adds new waypoints by clicking the add button for suggested waypoints and verifying the added waypoints.
+  async addNewWaypoints(
+    suggestedWaypointsCount: number,
+    waypointToAddNames: string[],
+    expectedValues: { name: string; ch: string; uic: string; km: string }[]
+  ) {
+    await this.addWaypointsButton.click();
+    await expect(this.viaModal).toBeVisible();
+    await expect(this.waypointSuggestions).toHaveCount(suggestedWaypointsCount);
+
+    let waypointSuggestionCount: number = 0;
+
+    while (waypointSuggestionCount < expectedValues.length) {
+      const waypointSuggestion = this.waypointSuggestions.nth(waypointSuggestionCount);
+      const expectedValue = expectedValues[waypointSuggestionCount];
+
+      await RoutePage.validateWaypointSuggestions(
+        waypointSuggestion,
+        expectedValue.name,
+        expectedValue.ch,
+        expectedValue.uic,
+        expectedValue.km
+      );
+
+      waypointSuggestionCount += 1;
+    }
+
+    await this.clickOnViaAddButtons(...waypointToAddNames);
+    await this.closeViaModalButton.click();
+
+    let droppedWaypointCount: number = 0;
```

Comment:
```suggestion
    let droppedWaypointCount = 0;
```

not really an issue, but I've seen this kind of typing at many places in this PR, doesn't it seem useless when you give a specific value?

Reply:
True